core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/util/DatabaseUtil.java

```diff
@@ -419,22 +426,19 @@ public static void udpateUserRoleMappingInBatchModeForInternalRoles(Connection d
                 log.debug(errorMessage, e);
             }
             throw new UserStoreException(errorMessage, e);
-        } finally {
+        } finally {  // can remove since this is not needed with try-with-resources -
             if (localConnection) {
                 DatabaseUtil.closeAllConnections(dbConnection);
```

Comment:
The try-with-resource pattern automatically handles the ResultSet and PreparedStatement objects, but the Connection, "dbConnection" in this scope, still needs to be closed.